xdsl/ir/core.py

```diff
@@ -614,6 +614,14 @@ class Operation(IRNode):
     def parent_node(self) -> IRNode | None:
         return self.parent
 
+    @property
+    def results_types(self) -> Sequence[Attribute]:
+        return tuple(r.type for r in self.results)
+
+    @property
+    def operands_types(self) -> Sequence[Attribute]:
+        return tuple(operand.type for operand in self.operands)
```

Comment:
is this following a convention? I would expect `result_types` and `operand_types`, even though the original names are plurals. This follows `result_types` as the argument name to `build`, and `arg_types` in the Block init.

Reply:
Can do!

Reply:
Done!